connector/proto/pom.xml

```diff
@@ -0,0 +1,119 @@
+<?xml version="1.0" encoding="UTF-8"?>
```

Comment:
Some top-level project related questions:

* (A) Do you plan to add Python support? 
    * The approach using descriptor files here keeps Python support simpler I think.
    * Overwhelming majority of the applications would be written in python.  
* (B) Do you plan to add scheme-registry support similar to how Avro does it?
    *  This is also a common use case for  Spark users.
* (C) Are maps supported?
     * It is also a common use case. 
* (D) Is there a restriction on protobuf v2 vs v3? Can users provide in either version?

Reply:
A) Yes, we are working on that. We wanted to get the initial review before starting on that.
B) We have done an initial prototype with Confluent schema registry. Is there a open source schema registry that you had in mind ?
C) Maps is supported
D) We have tested with proto3. Do you think proto2 support is important ?

Reply:
A) Great. Given how widely Python is used over scala, it would be essential. We could get this merged sooner than later and have Python support added. 
B) Confluent schema-registry is good. That is the most prevalent one. 
C) Map support: Is that yet to to be pushed to this github? I don't see it here.
D) proto 2 is very common. We should discuss about where it actually matters for this PR. Likely with minimal changes we could be version agnostic.

Reply:
B) Protobuf serialization with confluent schema registry adds  schemaId, MAGIC_BYTE etc. to the message. Don't we want to keep that out of open source ?
C) Map comes in as Message type. We do have a test case in ProtoFunctiosSuite. For example, a map with string type as "keys" and integer type as values will look like this when we deserialize:

StructField(map_name_age,
      ArrayType(
          StructType(
              StructField(key,StringType,true),
              StructField(value,IntegerType,true)
           ),false),
      false)
C) Okay, we will look into this and come back with a proposal.

Reply:
B) Avro handles it. Protobuf can too. I am wondering if we want to have an option not to expect the magic byte header. We can default to what Avro does (which expects magic_byte I think).

C) Ideally A field defined as Map should be a Map<> in Spark. That is what I had in mind with Map support. Current way to treat it as an array might be fine, but users would have to change their scripts when we have proper map support.

Reply:
B) This support is not there in the open source version unless I am mistaken. Also, we need confluent jars to talk to schema registry. Could you clarify what you mean ?
C) When deserializing a protobuf message, there is nothing like "Map" type. See [here](https://developers.google.com/protocol-buffers/docs/reference/java/com/google/protobuf/Descriptors.FieldDescriptor.JavaType.html). May be we are missing something. Can you clarify ?

Reply:
(B): Lets look into more details in the PR that adds schema-registry support. Yes, we would have to include the required jars.
(C) Lets discuss in my comment for SimpleMessageMap proto used in the tests below. Essentially a map in proto definition should ideally be a `MapType` in Spark.